src/particle_header.F90

```diff
@@ -154,4 +161,50 @@ elemental subroutine reset_coord(this)
 
   end subroutine reset_coord
 
+!===============================================================================
+! INITIALIZE_FROM_SOURCE initializes a particle from data stored in a source
+! site. The source site may have been produced from an external source, from
+! fission, or simply as a secondary particle.
+!===============================================================================
+
+  subroutine initialize_from_source(this, src)
+    class(Particle), intent(inout) :: this
+    type(Bank),      intent(in)    :: src
+
+    ! set defaults
+    call this % initialize()
+
+    ! copy attributes from source bank site
+    this % wgt            = src % wgt
+    this % last_wgt       = src % wgt
+    this % coord(1) % xyz = src % xyz
+    this % coord(1) % uvw = src % uvw
+    this % last_xyz       = src % xyz
+    this % last_uvw       = src % uvw
+    this % E              = src % E
+    this % last_E         = src % E
+
+  end subroutine initialize_from_source
+
+!===============================================================================
+! CREATE_SECONDARY stores the current phase space attributes of the particle in
+! the secondary bank and increments the number of sites in the secondary bank.
+!===============================================================================
+
+  subroutine create_secondary(this, uvw, type)
```

Comment:
Okay, I see.  I imagine we'll have to add an argument for energy at some point, right?  You can leave it for a later PR though, if you like.